### ISSUE TITLE
MAF difficult to train with complex target density

### ISSUE THREAD

**Issue by sachaguer** [sachaguer/jaxili#1]
In the usecase 'slcp' of sbi benchmark, MAFs are difficult to train and can return nan after a few training iterations. The initialisation is usually different from the target density leading to large losses at first iterations and noisy gradients.
Tuning hyperparameters such as the learning rate, the weight decay or the number of layers has been tries but didn't lead to conclusive results.
Further methods, like dropout layers or BatchNorm layers could be implemented to tackle this issue.
